spdb/spatialdb/annocube.py

```diff
@@ -92,6 +92,35 @@ def overwrite(self, input_data, time_sample_range=None):
             self.data[time_sample_range[0], :, :, :] = ndlib.overwriteDense64_ctype(
                 self.data[time_sample_range[0], :, :, :], input_data[time_sample_range[0], :, :, :])
 
+    def overwrite_to_black(self, input_data, time_sample_range=None):
+        """ Overwrite data with zero values in the input_data
+
+        If time_sample_range is provided, data will be inserted at the appropriate time sample
+
+        Args:
+            input_data (numpy.ndarray): Input mask matrix to overwrite the current Cube data
+            time_sample_range list(int): The min and max time samples that input_data represents in python convention
+            (start inclusive, stop exclusive)
+
+        Returns:
+            None
+
+        """
+        if self.data.dtype != input_data.dtype:
+            raise SpdbError("Conflicting data types for overwrite.",
+                            ErrorCodes.DATATYPE_MISMATCH)
+
+        if not time_sample_range:
+            # If no time sample range provided use default of 0
+            time_sample_range = [0, 1]
+
+        if input_data.ndim == 4:
+            for t in range(*time_sample_range):
+                self.data[t, :, :, :][input_data[t - time_sample_range[0], :, :, :]==1] = 0
+        else:
+            # Input data doesn't have any time indices
+            self.data[time_sample_range[0], :, :, :][input_data[time_sample_range[0], :, :, :]==1] = 0
```

Comment:
Is this correct?  The syntax doesn't look right to me. You want to assign all voxels in `self.data` to zero where you have a 1 in the `input_data`. This doesn't look like I'd expect. This comment applies to both the if and else blocks in all the overwrite_to_black methods.

Reply:
Currently what this line of code does is that it uses boolean indexing with the input data as a mask of `True`/`False` values. It scans through every voxel and if input data equals `1` it will replace that voxel with 0 in the self.data.

Reply:
Not sure if that explanation makes sense so here is a block of code that kind of shows what I'm going for here:

![Screen Shot 2019-12-19 at 3 04 07 PM](https://user-images.githubusercontent.com/39000785/71205679-e3b20e00-2270-11ea-8210-1b5b2af0c171.png)

Reply:
Then, from your example, shouldn't it be: `self.data[inputData[time_sample_range[0], :, :, :] == 1] = 0`?

Reply:
I think that would result in an error because `self_data` is a 4-dim array regardless if it has time indices and `input data[time_sample_range[0], :, :, :]` would always be a 3-dim array. So technically what its doing is getting a 3-dim subarray from `self.data[time_sample_range[0], :, :, :]` and then running the boolean indexing on that.

Reply:
Got it. Sorry, I use Numpy only very occasionally.